include/fcurl.h

```diff
@@ -29,6 +29,7 @@ size_t fcurl_read(void *ptr, size_t size, size_t nmemb, FCURL *stream);
 size_t fcurl_write(const void *ptr, size_t size, size_t nmemb,
                    FCURL *stream);
 int fcurl_flush(FCURL *stream);
+int fcurl_eof(FCURL *stream);
```

Comment:
This doesn't really "fix the read scenario" but is unrelated.

Reply:
Sorry, possibly mixed up with PR title that gets all the changes, the commit says "Provides a definition for fcurl_feof", which was missing. Should have made 2 PRs!